src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -164,7 +166,164 @@ pub fn update_lockfile(ws: &Workspace<'_>, opts: &UpdateOptions<'_>) -> CargoRes
     Ok(())
 }
 
-fn print_lockfile_updates(
+pub fn print_lockfile_changes(
+    gctx: &GlobalContext,
+    previous_resolve: Option<&Resolve>,
+    resolve: &Resolve,
+    registry: &mut PackageRegistry<'_>,
+) -> CargoResult<()> {
+    if let Some(previous_resolve) = previous_resolve {
+        print_lockfile_sync(gctx, previous_resolve, resolve, registry)
+    } else {
+        print_lockfile_generation(gctx, resolve, registry)
+    }
+}
+
+fn print_lockfile_generation(
+    gctx: &GlobalContext,
+    resolve: &Resolve,
+    registry: &mut PackageRegistry<'_>,
+) -> CargoResult<()> {
+    let mut shell = gctx.shell();
+
+    let diff = PackageDiff::new(&resolve);
+    let num_pkgs: usize = diff.iter().map(|d| d.added.len()).sum();
+    if num_pkgs <= 1 {
+        // just ourself, nothing worth reporting
+        return Ok(());
+    }
+    shell.status("Locking", format!("{num_pkgs} packages"))?;
+
+    for diff in diff {
+        fn format_latest(version: semver::Version) -> String {
```

Comment:
I lean towards not doing so.

As I was developing it, I avoided the urge to consolidate them because I didn't know where this was leading and they might diverge.  They still *might* but its unclear enough that I wouldn't make a decision off of it.

However, the function is simple enough and coupled enough to the calling code that I feel like consolidating them harms the ability to read each function and understand it.  I view this like creating a function for the duplicated "Locking" messages, the "Adding"s, etc.  Creating individual functions for those, while technically reducing duplication, makes the code too abstract.  We could consolidate all of the printing output but I didn't see a good way without making god functions that have flags for every situation.

Reply:
Understood. Asked for making sure it is intentional.